hook/scriptwrapper.go

```diff
@@ -145,9 +145,9 @@ func NewScriptWrapper(opts ...scriptWrapperOpt) (*ScriptWrapper, error) {
 	//
 	// But if the shebang specifies something weird like Ruby 🤪
 	// the wrapper won't work. Stick to POSIX shells for now.
-	if shebang != "" && !shellscript.IsPOSIXShell(shebang) {
-		return nil, fmt.Errorf("hook starts with an unsupported shebang line %q", shebang)
-	}
+	// if shebang != "" && !shellscript.IsPOSIXShell(shebang) {
+	// 	return nil, fmt.Errorf("hook starts with an unsupported shebang line %q", shebang)
+	// }
```

Comment:
this doesn't _technically_ have to be commented out anymore - it's still an error if we try to scriptwrap a non-shell hook, but we've kind of lifted this logic out to [here](https://github.com/buildkite/agent/pull/2040/files?diff=unified&w=0#diff-70e82e7432e3056b7466bb3e4fe16874c12cfbe7bbb71a15ad9d1997217e94c9R313). Maybe it makes sense to keep this in as a defensive programming thing?

Reply:
We can also defend ourselves with a test. I think that's more robust, so would prefer to delete the comments.

Reply:
oh totally, if this gets out of WIP i'd either de-comment it or remove these lines entirely.

when you say we can defend ourselves with a test, do you mean adding a test that asserts that if we try to wrap a non-shell-script, it complains?

Reply:
Yeah, exactly. I's image, we would want to test a whole table of binaries and scripts and verify that they run. We should be able to tease out a test that fails when the scriptwrapper is run on the wrong type of binary from this.

You might have to think a bit about making an interface. I think that's a good thing to do, though.

Reply:
yep, that sounds like a good idea. when we get to properly testing this, i'll do that.

Reply:
Have we gotten to properly testing this?

Reply:
very good call! when i said:

> oh totally, if this gets out of WIP i'd either de-comment it or remove these lines entirely.

apparently i was lying :P

i'll bang out some tests for this now